### PR TITLE
Add com.sigil_ebook.Sigil

### PR DESCRIPTION
**Pull request by catsout**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:https://github.com/Sigil-Ebook/Sigil/issues/521**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by catsout** [comment]
bot, build com.sigil_ebook.Sigil

**Comment by catsout** [comment]
bot, build com.sigil_ebook.Sigil

File: com.sigil_ebook.Sigil.yml [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+rename-desktop-file: sigil.desktop
+rename-icon: sigil
+command: sigil
+finish-args:
+  - --socket=x11
+  - --socket=wayland
+  - --share=ipc
+  - --share=network
+  - --device=dri
+  - --filesystem=xdg-documents
```
Comment (barthalion) [inline com.sigil_ebook.Sigil.yml:16]: I can imagine it should be `home` as you cannot really know where someone stores their books.

File: com.sigil_ebook.Sigil.yml [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+runtime: org.kde.Platform
+runtime-version: 5.15-21.08
+sdk: org.kde.Sdk
+base: io.qt.qtwebengine.BaseApp
+base-version: 5.15-21.08
+rename-desktop-file: sigil.desktop
+rename-icon: sigil
+command: sigil
+finish-args:
+  - --socket=x11
```
Comment (barthalion) [inline com.sigil_ebook.Sigil.yml:11]:
```suggestion
  - --socket=fallback-x11
```

File: dependencies/sigil-pypi-dependencies.json [hunk 3]
```diff
@@ -0,0 +1,134 @@
+{
+    "name": "sigil-pypi-dependencies",
+    "buildsystem": "simple",
+    "build-commands": [],
+    "modules": [
+        {
+            "name": "python3-lxml",
+            "buildsystem": "simple",
+            "build-commands": [
+                "pip3 install --verbose --exists-action=i --no-index --find-links=\"file://${PWD}\" --prefix=${FLATPAK_DEST} \"lxml\" --no-build-isolation"
```
Comment (hfiguiere) [inline dependencies/sigil-pypi-dependencies.json:10]: you need to add a `--ignore-installed`

for some reason python3-xml is in the SDK but not the platform.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: com.sigil_ebook.Sigil.yml (hunk 1 above)
Comment (hfiguiere) [reply com.sigil_ebook.Sigil.yml:16]: since it's an editor, it's to save them. isn't it?

File: dependencies/sigil-pypi-dependencies.json (hunk 3 above)
Comment (catsout) [reply dependencies/sigil-pypi-dependencies.json:10]: Seems with `--exists-action=i --prefix=${FLATPAK_DEST}`, it already ignores installed.

**Comment by catsout** [comment]
bot, build com.sigil_ebook.Sigil

**Pull request closed without merging**